resources/aws_ec2.go

```diff
@@ -486,6 +491,69 @@ func (obj *AwsEc2Res) longpollWatch() error {
 	}
 }
 
+// snsWatch uses amazon cloudwatch events and simple notification service to
+// detect ec2 instance state changes.
+func (obj *AwsEc2Res) snsWatch() error {
+	send := false
+	var exit *error
+	defer obj.wg.Wait()
+	defer close(obj.closeChan)
+	// set up the sns endpoint
+	snsServer := obj.snsServer()
+	// shutdown the sns endpoint when we're done
+	defer func() {
+		ctx, _ := context.WithTimeout(context.TODO(), SnsServerTimeout*time.Second)
+		if err := snsServer.Shutdown(ctx); err != nil {
+			log.Printf("%s: error stopping sns endpoint: %s", obj, err)
+			return
+		}
+	}()
+	obj.wg.Add(1)
+	// start the endpoint
+	go func() {
+		defer obj.wg.Done()
+		defer close(obj.awsChan)
+		if err := snsServer.ListenAndServe(); err != nil {
+			// when we shut down
+			if err == http.ErrServerClosed || err == context.Canceled {
```

Comment:
@purpleidea: I finished the changes, but wanted to check two things with you about this error.
1. Is the error checking correct? as I understand it http returns the context error (context.Canceled) if it's there.
2. I think this is the correct place to print the "Stopped SNS Endpoint" message, even though it's more logical in the defer func at first glance, but would print even if there was an http error. What do you think?

Reply:
I think I know the answer to 2 and 3, but having some insecurity :P
1 ?
2 yes
3 this is good